Stage 1 Submission

Pavkv opened the pull request.

Pavkv committed:
Initial Submission

Pavkv committed:
Update README.md

ImaliaNsk commented:
Thank you for sending your code. The project looks pretty good in general, however, it will require some refactoring. First of all, these issues must be fixed before the next review:

- The project doesn't run after "npm run dev": https://prnt.sc/o4bwpXjQ2O63 Please fix scripts in your `package.json`
- The deployment link in the README leads to 404, not found.  

And there are also some other issues to work on.

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.

* The content of the page should have limited width. Currenlty, on large screen resolutions, it's stretched to the whole width (e.g. on 2800px the content will occupy the whole screen width). Backgrounds can stretch to the whole width. But the content itself must be limited like in figma
* https://prnt.sc/sD2rc1V3K9i1 Active link should be underlined
* Modals shuold be centered vertically: https://prnt.sc/3ynedrJmXeLo
* "Registration" successful modal is missing
* With the search keyword entered, the input looks incorrectly, without rounded corners: https://prnt.sc/1PabYAF6f61G

- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.

* saved-news page scrolls on resolutions 615 and up
* home page scrolls on all resolutions

- [ ] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.

* Links in the footer should be opening in a new tab.

- [ ] <!--4.2--> Classes are named according to BEM specifications.

* https://prnt.sc/yQ3fBzZnhiye Modifiers like these should be key-value, e.g. `_theme_light`. Please correct all similar cases.
* https://prnt.sc/J5B443ZQFuQT Incorrectly used modifier. Please correct all similar cases.

- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.

* `main`, `section`, `footer` are missing
* Please don't use `br` to divide text into paragraphs (section `about`). The text semantically consists of two paragraphs, not one paragraph. `br` is a presentation only tag, shouldn't be used in such cases.

- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using Vite or CRA.
- [ ] <!--4.2--> The popup windows have opening and closing functionality implemented.

* "Registration" successful modal is missing

- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [ ] <!--4.2--> There are no linting errors or warnings.

* The project doesn't run after "npm run dev": https://prnt.sc/o4bwpXjQ2O63 Please fix scripts in your `package.json`

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.

* Hover effect on trash can icon from the mockup: https://prnt.sc/KFTCWRfN-fBQ

- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [ ] <!--1.66--> Raster and vector images have been optimized.

* It's better to reduce the size of images. Using optimized images will save bandwidth and result in faster website loading times.

## Number of points: 41

Pavkv committed:
fix: The content of the page with limited width 1440px;
fix: Underline active link;
fix: Center modals vertically;
feat: "Registration" successful modal;
fix: Remove overflow in mobile view;
fix: Input looks with the search keyword entered;
fix: Open links in the footer in a new tab;
fix: Classes are named according to BEM specifications;
feat: Add header, main and footer sections;
fix: Hover effect on trash can icon;
fix: Optimize raster and vector images
